sglang/code-walk-through/readme.md

```diff
@@ -46,8 +46,8 @@ Note that all the discussions are based on release [v0.4.0](https://github.com/s
 SGLang features an SRT (SGLang Runtime) Server for [serving online HTTP requests](https://sgl-project.github.io/start/send_request.html) and an Engine for [offline model execution](https://sgl-project.github.io/backend/offline_engine_api.html). Key functions, [`launch_server`](https://github.com/sgl-project/sglang/blob/f8b0326934bacb7a7d4eba68fb6eddebaa6ff751/python/sglang/srt/server.py#L507) and [`launch_engine`](https://github.com/sgl-project/sglang/blob/f8b0326934bacb7a7d4eba68fb6eddebaa6ff751/python/sglang/srt/server.py#L418), are in [server.py](https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/server.py). The `launch_engine` function initializes core SRT Server components.
 
 1. Set up configs (logger, server args, CUDA/NCCL env, inter-process ports) and download the model and tokenizer.
-2. Run Scheduler processes: Each Scheduler runs TpModelWorker for prefill and decode, manage radix cache, and handles TokenizerManager requests in infinite event loop. If `dp_size > 1`, `run_data_parallel_controller_process`; otherwise, initialize a Scheduler for each `tp_rank`.
-3. Run TokenizerManager and DetokenizerManager as subprocesses: the former tokenizes data for the Scheduler, and the latter detokenizes Scheduler outputs for the server frontend. For multi-node inference (e.g., Llama 3.1 405B), TokenizerManager and DetokenizerManager only run on the first node.
+2. Run Scheduler processes: Each Scheduler runs TpModelWorker for prefill and decode, manage Radix Cache, and handles TokenizerManager requests in infinite event loop. If `dp_size > 1`, `run_data_parallel_controller_process`; otherwise, initialize a Scheduler for each `tp_rank`.
+3. Run TokenizerManager and DetokenizerManager as subprocesses: the former tokenizes data for the Scheduler, and the latter detokenizes Scheduler outputs for the server frontend. For multi-node inference (e.g., Llama 3.1 405B with TP8), TokenizerManager and DetokenizerManager only run on the first node.
```

Comment:
The mention of "multi-nodes" here could be misinterpreted as compute nodes with 8 GPUs rather than just 8 GPUs, so adding "with TP8" helps clarify that we're referring to Tensor Parallel across 8 GPUs.

Reply:
I think multi-node in MLSys just means "compute nodes with 8 GPUs". This is not TP across 8 GPUs. Even traditionally, machines that takes network trips to access one another are called nodes. The 8-GPUs can access each other through PCIE/NVLink and share CPUs, that's why they are called a "node". @BBuf @zhaochenyang20

Reply:
Okay, I agree with this point, I'll revert this change.

Reply:
Sure!